File: sdks/js/src/opspec/interpreter/opcall/params/param/array/validate.ts
import Ajv from 'ajv'
const ajv = new Ajv()

/**
* validates value against constraints
* @param {Array} value
* @param {Object} [constraints]
* @return {Array<Error>}
*/
export default function validate(
 value: any[],
 constraints?: any
) {
 constraints = Object.assign({ type: 'array' }, constraints)

 ajv.validate(constraints, value)
 return ajv.errors || []
}

File: sdks/go/opspec/interpreter/call/op/params/param/file/validate.go
package file

import (
	"errors"
	"fmt"
	"os"

	"github.com/opctl/opctl/sdks/go/model"
)

// Validate validates a value against a string parameter
func Validate(
	value *model.Value,
) []error {
	if value == nil || value.File == nil {
		return []error{errors.New("file required")}
	}

	fileInfo, err := os.Stat(*value.File)
	if err != nil {
		return []error{err}
	} else if !fileInfo.Mode().IsRegular() {
		return []error{fmt.Errorf("%v not a file", *value.File)}
	}
	return []error{}
}
